### PR TITLE
Fix parsing of boolean value

### LINKED ISSUE #123: Inconsistent parsing of `bool`
https://github.com/andrey-zherikov/argparse/blob/7bfd8022023cfd3cb6374c2d14c50f1895f3bc45/source/argparse/internal/valueparser.d#L149-L160

*yes*, *y*, *no*, and *n* are accepted in lower case only, but *true* and *false* are matched case-insensitively by `std.conv.to`. Also, when a parsing error occurs, the message only mentions *true*/*false* as allowed values.

### PR DESCRIPTION
**Pull request by andrey-zherikov**
Closes #123

### CODE REVIEW

**Comment by SirNickolas** [comment]
I’ll seize the opportunity to thank you for *not* supporting `on`/`off`. Having both `on` and `no` in the same domain of values is a footgun.

**Comment by andrey-zherikov** [comment]
This can be made configurable but since there are no complains, it's not urgent.

File: source/argparse/internal/valueparser.d [hunk 1]
```diff
@@ -139,19 +139,18 @@ if(!is(T == void))
...
         alias DefaultValueParser = ValueParser!(
-            void,                               // pre process
-            void,                               // pre validate
+            (ref RawParam param)                // pre process
+            {
+                import std.string: toLower;
+                import std.algorithm: each;
+
+                // convert values to lower case and replace "" with "y"
+                param.value.each!((ref _) { _ = _.length == 0 ? "y" : _.toLower; });
```
[the remark this answers is not in the document]
Comment (andrey-zherikov) [reply source/argparse/internal/valueparser.d:148]: What's the benefit of this comparing to my version?

File: source/argparse/internal/valueparser.d [hunk 2]
```diff
@@ -139,19 +139,18 @@ if(!is(T == void))
...
-                switch(value)
-                {
-                    case "":    goto case;
-                    case "yes": goto case;
-                    case "y":   return true;
-                    case "no":  goto case;
-                    case "n":   return false;
-                    default:    return value.to!T;
-                }
+                return ["true":true, "yes":true, "y":true].get(value, false);
```
[the remark this answers is not in the document]
Comment (andrey-zherikov) [reply source/argparse/internal/valueparser.d]: Makes sense.

File: source/argparse/internal/valueparser.d (hunk 1 above)
[the remark this answers is not in the document]
Comment (SirNickolas) [reply source/argparse/internal/valueparser.d:148]: Sorry for disappearing for so long.

The benefit is in doing less work. You used [`std.uni.toLower`][uni] (publicly imported into `std.string`), and this version uses [`std.ascii.toLower`][ascii]. When the character being processed is ASCII, the former performs the same actions as the latter; when not, it has to consult Unicode tables (and they are huge).

We don’t really care here whether non-ASCII characters will get lowercased – after all, we are comparing the value against ASCII-only strings. So a simpler function fulfills our needs.

[uni]: https://github.com/dlang/phobos/blob/v2.106.0/std/uni/package.d#L9975
[ascii]: https://github.com/dlang/phobos/blob/v2.106.0/std/ascii.d#L647

**Review by SirNickolas — APPROVED** [review]

**Pull request merged**